index.js

```diff
@@ -106,6 +118,11 @@
     return f(this.value);
   };
 
+  //  Just#concat :: m a -> m a
+  Just.prototype.concat = function(maybe) {
+    return maybe instanceof Just ? Just(this.value.concat(maybe.value)) : this;
+  };
+
```

Comment:
> Sorry, I think my remark was a little snarky. That wasn't my intention.

I didn't sense snark.  I sensed us likely talking past each other.  Most likely confusion, although perhaps some real fundamental disagreement too.  It's hard because I have a mix of strong backgrounds in certain of the underlying topics and weak ones in others -- but strong opinions about everything, I'm afraid!

> My qualm is just that the other functions don't do any checking, but you want `concat` to do a bunch. I don't understand why `concat` needs to be special. 

It's more basic than that for me.  I simply don't see a reason to include `concat`.  I haven't seen any good argument that the sort of `Maybe` so far defined should be a `Semigroup` or a `Monoid`.  And because of the types of issues discussed, I see real issues with trying to define it that way.

But I think even further down is the notion that I suspect you don't think about `Maybe` as a fully-fledged type.  Am I right?  Do you think in terms of `Maybe a` for some other type `a`?  To me, the `Maybe` type as so far defined by Sanctuary, is a type on its own as far as I can tell.  And that might explain a lot of our disagreement.  Such a type as I see of course cannot comfortably be made part of a `Semigroup`.  Did I get your position right?  Does this help explain our differences?  If so, do you want to try to convince me that `Maybe`, regardless of current implementation, should only be thought of as `Maybe a`?